src/Umbraco.Core/Persistence/Migrations/Upgrades/TargetVersionSevenTenZero/RenamePreviewFolder.cs

```diff
@@ -0,0 +1,41 @@
+using System.IO;
+using Umbraco.Core.IO;
+using Umbraco.Core.Logging;
+using Umbraco.Core.Persistence.SqlSyntax;
+using File = System.IO.File;
+
+namespace Umbraco.Core.Persistence.Migrations.Upgrades.TargetVersionSevenTenZero
+{
+    /// <summary>
+    /// Renames the preview folder containing static html files to ensure it does not interfere with the MVC route
+    /// that is now supposed to render these views dynamically. We don't want to delete as people may have made
+    /// customizations to these files that would need to be migrated to the new .cshtml view files.
+    /// </summary>
+    [Migration("7.10.0", 1, Constants.System.UmbracoMigrationName)]
+    public class RenamePreviewFolder : MigrationBase
+    {
+        public RenamePreviewFolder(ISqlSyntaxProvider sqlSyntax, ILogger logger)
+            : base(sqlSyntax, logger)
+        {
+        }
+
+        public override void Up()
+        {
+            var previewFolderPath = IOHelper.MapPath(SystemDirectories.Umbraco + "/preview");
+            if (Directory.Exists(previewFolderPath))
+            {
+                var newPath = previewFolderPath.Replace("preview", "preview.old");
+                Directory.Move(previewFolderPath, newPath);
```

Comment:
Turns out this causes a critical error during upgrades if it's already been copied, there's no check here for whether the preview.old already exists. I'll fix

Reply:
Ouch, good point, thanks!